plutus-pab/src/Plutus/PAB/Db/Eventful/ContractStore.hs

```diff
@@ -36,13 +36,22 @@ handleContractStore = \case
         void
             $ runCommand @() @(PABEvent (ContractDef t))
                 Command.updateContractInstanceState
-                PABEventSource
+                (InstanceEventSource i)
```

Comment:
This is here because you want to be able to log the instance id in the simulator?

Reply:
no I want to have the instance ID in the `Source` so that we can use it for filtering the events in our `eventful` queries later on. At the moment every query works on all events, so I'm hoping that we can add some sharding by using the instance ID for instance-specific events. (I haven't found the right API functions in `eventful` for the queries yet :( )